kong/api/routes/upstreams.lua

```diff
@@ -40,7 +42,48 @@ return {
     end,
 
     GET = function(self, dao_factory)
-      crud.paginated_set(self, dao_factory.targets)
+      -- by default, just get a paginated list
+      if not self.params.active then
+        return crud.paginated_set(self, dao_factory.targets)
+      end
+
+      self.params.active = nil
+
+      -- user wanted a list of all active targets for this upstream
+      local target_history, err = dao_factory.targets:find_all({
+        upstream_id = self.params.upstream_id,
+      })
+
+      if not target_history then
+        return app_helpers.yield_error(err)
+      end
+
+      -- sort and walk based on target and creation time
+      for _, target in ipairs(target_history) do
+        target.order = target.target..":"..target.created_at
+      end
+      table.sort(target_history, function(a, b) return a.order > b.order end)
```

Comment:
(Also I would contend that the argument that each sort result must be identical is invalid; in this case, what we care about most is the most recent entry of a target, because what we're searching for is the result of a target's most recent entry's weight being 0 or not)